### PR TITLE
Feature/proj 194

### PR DESCRIPTION
**Pull request by EvgeniyaYakunina**
This pull request introduces two new components, Notifications and NotificationItem, to handle and display notifications within the application. Here's a breakdown of what each component does:

### 1. Notifications Component
The Notifications component is responsible for displaying a dropdown menu containing all notifications. It uses the useGetAllNotificationsQuery hook to fetch notifications and the useConnectSocket hook to track new (unread) notifications in real-time. The component maintains the state for the number of new notifications and controls the dropdown menu's open/close state based on user interaction. It renders a list of NotificationItem components for each notification in the dropdown menu.

### 2. NotificationItem Component

The NotificationItem component displays the details of a single notification. It takes a notification prop of type Notification and renders the following information:

A read status indicator (a tag displaying "New" for unread notifications).
The notification message.
The time elapsed since the notification was created, formatted using the useTimeAgo hook.
All text is internationalized using the useTranslation hook.

### Additional Changes:

Created a **SocketApi** class to handle Socket.IO connections and events. It establishes a connection to the Socket.IO server, listens for 'connect', 'disconnect', and 'connect_error' events, and emits a 'disconnect' event when the connection is closed.
Created a useConnectSocket hook that initializes a Socket.IO connection, listens for notifications, and updates the notifications state with the received data. It also cleans up the connection when the component unmounts.
notif

### CODE REVIEW

**Review by NoName-ForTeam — APPROVED** [review]

File: src/features/profile/api/devices/device.api.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { baseApi } from '@/app/api'
-import { GetSessions } from '@/features/device/api/device.types'
+import { GetSessions } from '@/features/profile/api/devices/device.types'
```
Comment (Lelik7777) [inline src/features/profile/api/devices/device.api.ts]: длинный путь

File: src/features/profile/ui/notifications/notification/notification.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <>
+      <div>
+        {notification.isRead ? (
+          ''
+        ) : (
+          <div className={styles.notify}>
+            <Typography variant={'textBold14'}>{t.notifications.newNotification}</Typography>
+            <span className={styles.new}>{t.notifications.new}</span>
```
Comment (Lelik7777) [inline src/features/profile/ui/notifications/notification/notification.tsx]: почему также не использовать типографию?

File: src/features/profile/ui/notifications/notificationsIcon/notificationsIcon.module.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  width: 13px;
+  height: 13px;
+
+  font-size: 10px;
+  font-weight: var(--font-weight-medium);
+  line-height: 8px;
```
Comment (Lelik7777) [inline src/features/profile/ui/notifications/notificationsIcon/notificationsIcon.module.scss]: use mixin font

File: src/features/profile/ui/notifications/notificationsIcon/notificationsIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  newNotifications: number
+}
+export const NotificationsIcon = ({ isOpen, newNotifications }: NotificationsIconProps) => (
+  <div className={styles.icons}>
+    {isOpen ? (
+      <FillBell className={styles.activeBell} />
+    ) : (
+      <OutlineBell className={styles.defaultBell} />
+    )}
+    {newNotifications > 0 && <span className={styles.count}>{newNotifications}</span>}
```
Comment (Lelik7777) [inline src/features/profile/ui/notifications/notificationsIcon/notificationsIcon.tsx]: почему не использовать типографию?

File: src/shared/utils/hooks/useConnectSocket.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react'
+
+import { SocketApi } from '@/app/api'
+import { Notification } from '@/features'
+import { WS_EVENT_PATH } from '@/shared/config'
+
+export const useConnectSocket = () => {
+  const [notifications, setNotifications] = useState<Notification>()
+
+  console.log('Notifications state:', notifications)
```
Comment (Lelik7777) [inline src/shared/utils/hooks/useConnectSocket.ts]: консоль

File: src/features/profile/api/devices/device.api.ts (hunk 1 above)
Comment (EvgeniyaYakunina) [reply src/features/profile/api/devices/device.api.ts]: поправила

File: src/shared/utils/hooks/useConnectSocket.ts (hunk 5 above)
Comment (EvgeniyaYakunina) [reply src/shared/utils/hooks/useConnectSocket.ts]: убрала

File: src/features/profile/ui/notifications/notificationsIcon/notificationsIcon.tsx (hunk 4 above)
Comment (EvgeniyaYakunina) [reply src/features/profile/ui/notifications/notificationsIcon/notificationsIcon.tsx]: Нет нужного комплекта параметров ни в одной typography

File: src/features/profile/ui/notifications/notification/notification.tsx (hunk 2 above)
Comment (EvgeniyaYakunina) [reply src/features/profile/ui/notifications/notification/notification.tsx]: Нет нужного комплекта параметров ни в одной typography

File: src/features/profile/ui/notifications/notificationsIcon/notificationsIcon.module.scss (hunk 3 above)
Comment (EvgeniyaYakunina) [reply src/features/profile/ui/notifications/notificationsIcon/notificationsIcon.module.scss]: Создала новый размер в typography.scss, добавила  mixin

**Review by Lelik7777 — APPROVED** [review]

**Comment by EvgeniyaYakunina** [comment]
run build production

**Pull request merged**